mailbox/event/event-rabbitmq/src/test/java/org/apache/james/mailbox/events/ReactorRabbitMQChannelPoolTest.java

```diff
@@ -0,0 +1,87 @@
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.events;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ExecutionException;
+
+import org.apache.james.backend.rabbitmq.RabbitMQExtension;
+import org.apache.james.util.concurrency.ConcurrentTestRunner;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import com.github.fge.lambdas.Throwing;
+import com.rabbitmq.client.Channel;
+
+import reactor.core.publisher.Mono;
+import reactor.rabbitmq.ChannelPool;
+
+class ReactorRabbitMQChannelPoolTest implements ChannelPoolContract {
+
+    @RegisterExtension
+    static RabbitMQExtension rabbitMQExtension = RabbitMQExtension.singletonRabbitMQ();
+
+    private List<ReactorRabbitMQChannelPool> channelPools;
+
+    @BeforeEach
+    void beforeEach() {
+        channelPools = new ArrayList<>();
+    }
+
+    @AfterEach
+    void afterEach() {
+        channelPools.forEach(ReactorRabbitMQChannelPool::close);
+    }
+
+    @Override
+    public ChannelPool getChannelPool(int poolSize) {
+        ReactorRabbitMQChannelPool channelPool = new ReactorRabbitMQChannelPool(
+            rabbitMQExtension.getRabbitConnectionPool().getResilientConnection(),
+            poolSize);
+        channelPools.add(channelPool);
+
+        return channelPool;
+    }
+
+    // Pool wait timeout is an expected exception
+    @Test
+    void concurrentRequestOnChannelMonoLeadToPoolWaitTimeoutException() {
+        Mono<? extends Channel> channelMono = getChannelPool(99).getChannelMono();
+
+        ConcurrentLinkedQueue<Channel> listChannels = new ConcurrentLinkedQueue<>();
+        assertThatThrownBy(() ->
+            ConcurrentTestRunner.builder()
+                .operation((threadNumber, step) -> listChannels.add(channelMono.block()))
+                .threadCount(10)
+                .operationCount(10)
+                .runSuccessfullyWithin(Duration.ofSeconds(30)))
+            .isInstanceOf(ExecutionException.class)
+            .hasMessageContaining("java.util.NoSuchElementException: Timeout waiting for idle object");
+
+        listChannels.forEach(Throwing.consumer(Channel::close));
```

Comment:
that's not ok: you should make sure channels are all closed every time (in finally or in @After) and you should catch exceptions to at least try to close each channel.